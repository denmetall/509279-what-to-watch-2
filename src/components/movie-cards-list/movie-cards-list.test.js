import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardsList from "./movie-cards-list.jsx";
import filmsMock from '../../mocks/films';
import {BrowserRouter as Router} from 'react-router-dom';
import createNodeMock from '../../mocks/create-node-mock';

it(`Components MovieCardsList renders correctly`, () => {
  const options = {createNodeMock};

  const tree = renderer
    .create(<Router>
      <MovieCardsList
        films={filmsMock}
        onMouseOverCard={jest.fn()}
        onMouseOutCard={jest.fn()}
        activeMovieId={filmsMock[0].id}
      />
    </Router>, options)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
