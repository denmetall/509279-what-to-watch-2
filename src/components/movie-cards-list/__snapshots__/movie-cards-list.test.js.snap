// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components MovieCardsList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/johnny-english.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/1"
        onClick={[Function]}
      >
        Drive
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/macbeth.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/2"
        onClick={[Function]}
      >
        Astral
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/johnny-english.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/3"
        onClick={[Function]}
      >
        Taxi
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/macbeth.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/4"
        onClick={[Function]}
      >
        Resident evil
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/mindhunter.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/5"
        onClick={[Function]}
      >
        Brother
      </a>
    </h3>
  </article>
</div>
`;
